src/api/routes/socket-io.ts

```diff
@@ -125,93 +124,100 @@ export function createSocketIORouter(db: DataStore, server: http.Server) {
     logger.info(`[socket.io] socket ${id} left room "${room}"`);
   });
 
-  db.on('blockUpdate', (dbBlock, txIds, microblocksAccepted, microblocksStreamed) => {
+  db.on('blockUpdate', async (blockHash, microblocksAccepted, microblocksStreamed) => {
     // Only parse and emit data if there are currently subscriptions to the blocks topic
     const blockTopic: Topic = 'block';
     if (adapter.rooms.has(blockTopic)) {
+      const dbBlockQuery = await db.getBlock({ hash: blockHash });
+      if (!dbBlockQuery.found) {
+        return;
+      }
+      const dbBlock = dbBlockQuery.result;
+      const dbTxsQuery = await db.getBlockTxs(blockHash);
+      const txIds = dbTxsQuery.results;
       const block = parseDbBlock(dbBlock, txIds, microblocksAccepted, microblocksStreamed);
       prometheus?.sendEvent('block');
       io.to(blockTopic).emit('block', block);
     }
   });
 
-  db.on('txUpdate', dbTx => {
+  db.on('txUpdate', async txId => {
     // Only parse and emit data if there are currently subscriptions to the mempool topic
     const mempoolTopic: Topic = 'mempool';
     if (adapter.rooms.has(mempoolTopic)) {
-      // only watch for mempool txs
-      if ('receipt_time' in dbTx) {
-        // do not send updates for dropped/pruned mempool txs
-        if (!dbTx.pruned) {
-          const tx = parseDbMempoolTx(dbTx);
-          prometheus?.sendEvent('mempool');
-          io.to(mempoolTopic).emit('mempool', tx);
-        }
+      const dbTxQuery = await db.getMempoolTx({
+        txId: txId,
+        includeUnanchored: true,
+        includePruned: false,
+      });
+      if (!dbTxQuery.found) {
+        return;
       }
+      const dbMempoolTx = dbTxQuery.result;
+      const tx = parseDbMempoolTx(dbMempoolTx);
+      prometheus?.sendEvent('mempool');
+      io.to(mempoolTopic).emit('mempool', tx);
     }
   });
 
-  db.on('addressUpdate', info => {
-    // Check for any subscribers to tx updates related to this address
-    const addrTxTopic: AddressTransactionTopic = `address-transaction:${info.address}` as const;
+  db.on('addressUpdate', async (address, blockHeight) => {
+    const addrTxTopic: AddressTransactionTopic = `address-transaction:${address}` as const;
+    const addrStxBalanceTopic: AddressStxBalanceTopic = `address-stx-balance:${address}` as const;
+    if (!adapter.rooms.has(addrTxTopic) && !adapter.rooms.has(addrStxBalanceTopic)) {
+      return;
+    }
+    const dbTxsQuery = await db.getAddressTxsWithAssetTransfers({
+      stxAddress: address,
+      limit: 20,
```

Comment:
@zone117x what do you think is a sensible limit to place here? Ideally I'd like to get all the asset transfers in this tx

Reply:
When this query is limited to a single block, seems reasonable to fetch all txs